### PR TITLE
[Move] Implement (or re-implement?) Lucky Chant

### PR DESCRIPTION
**Pull request by innerthunder**
## What are the changes?

This fully implements the move [Lucky Chant](https://bulbapedia.bulbagarden.net/wiki/Lucky_Chant_(move)).

## Why am I doing these changes?

Lucky Chant is unimplemented... or, at least, it's marked as unimplemented. Before these changes, Lucky Chant used a battler tag to prevent critical hits on the user, but since it was marked (N), its effects were not functional and the move was disabled. Also, just removing the `.unimplemented()` flag from the move was insufficient since Lucky Chant is meant to protect both the user and its ally. The changes in this PR bring Lucky Chant to its full functionality.

## What did change?

- `data/arena-tag` and `enums/arena-tag-type`: New `NoCritTag` (of tag type `ArenaTagType.NO_CRIT`) prevents hits against the tag's side from dealing critical damage.
- `data/battler-tags` and `enums/battler-tag-type`: Removed `BattlerTagType.NO_CRIT`, a part of Lucky Chant's previous partial implementation.
- `data/move`: Updated Lucky Chant:
  - Removed `.unimplemented()` flag
  - Changed `AddBattlerTagAttr` to `AddArenaTagAttr` with the new `NO_CRIT` arena tag.
- `field/pokemon`: Minor changes reflecting changes to `NO_CRIT` tag implementation.
- `test/moves/lucky_chant.test`: New integration test for Lucky Chant's effects in single and double battles
- `test/moves/dragon_rage.test`: Fixed a test that used the `NO_CRIT` tag incorrectly.

### Screenshots/Videos

https://github.com/user-attachments/assets/b7abd7b6-7f25-4e45-a98e-3cd938d43fc8

## How to test the changes?

`npm run test lucky_chant`

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by innerthunder** [comment]
TODO before moving out of draft:
- Localize messages in `NoCritTag`
- Record test video

File: src/locales/ko/arena-tag.ts [hunk 1]
```diff
@@ -22,6 +22,9 @@ export const arenaTag: SimpleTranslationEntries = {
   "conditionalProtectOnAddEnemy": "상대 주변을\n{{moveName}}[[가]] 보호하고 있다!",
   "conditionalProtectApply": "{{pokemonNameWithAffix}}[[를]]\n{{moveName}}[[가]] 지켜주고 있다!",
   "matBlockOnAdd": "{{pokemonNameWithAffix}}[[는]]\n마룻바닥세워막기를 노리고 있다!",
+  "noCritOnAddPlayer": "The {{moveName}} shielded your\nteam from critical hits!",
+  "noCritOnAddEnemy": "The {{moveName}} shielded the opposing\nteam from critical hits!",
+  "noCritOnRemove": "{{pokemonNameWithAffix}}'s {{moveName}}\nwore off!",
```
Comment (EnochG1) [inline src/locales/ko/arena-tag.ts:27]:
```suggestion
  "noCritOnAddPlayer": "{{moveName}}의 힘으로\n우리 편의 급소가 숨겨졌다!",
  "noCritOnAddEnemy": "{{moveName}}의 힘으로\n상대의 급소가 숨겨졌다!",
  "noCritOnRemove": "{{pokemonNameWithAffix}}의 {{moveName}}[[가]] 풀렸다!",
```

File: src/locales/pt_BR/arena-tag.ts [hunk 2]
```diff
@@ -22,6 +22,9 @@ export const arenaTag: SimpleTranslationEntries = {
   "conditionalProtectOnAddEnemy": "{{moveName}} protegeu a\nequipe adversária!",
   "conditionalProtectApply": "{{moveName}} protegeu {{pokemonNameWithAffix}}!",
   "matBlockOnAdd": "{{pokemonNameWithAffix}} pretende levantar um tapete\npara bloquear ataques!",
+  "noCritOnAddPlayer": "The {{moveName}} shielded your\nteam from critical hits!",
+  "noCritOnAddEnemy": "The {{moveName}} shielded the opposing\nteam from critical hits!",
+  "noCritOnRemove": "{{pokemonNameWithAffix}}'s {{moveName}}\nwore off!",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/arena-tag.ts:27]:
```suggestion
  "noCritOnAddPlayer": "{{moveName}} protegeu sua\equipe de acertos críticos!",
  "noCritOnAddEnemy": "{{moveName}} protegeu a\equipe adversária de acertos críticos",
  "noCritOnRemove": "{{moveName}} de {{pokemonNameWithAffix}}\nacabou!",
```

File: src/locales/fr/arena-tag.ts [hunk 3]
```diff
@@ -22,6 +22,9 @@ export const arenaTag: SimpleTranslationEntries = {
   "conditionalProtectOnAddEnemy": "La capacité {{moveName}}\nprotège l’équipe ennemie !",
   "conditionalProtectApply": "{{pokemonNameWithAffix}} est protégé\npar {{moveName}} !",
   "matBlockOnAdd": "{{pokemonNameWithAffix}} se prépare\nà utiliser un tatami pour bloquer les attaques !",
+  "noCritOnAddPlayer": "The {{moveName}} shielded your\nteam from critical hits!",
+  "noCritOnAddEnemy": "The {{moveName}} shielded the opposing\nteam from critical hits!",
+  "noCritOnRemove": "{{pokemonNameWithAffix}}'s {{moveName}}\nwore off!",
```
Comment (Adri1) [inline src/locales/fr/arena-tag.ts:27]:
```suggestion
  "noCritOnAddPlayer": "{{moveName}} immunise votre équipe\ncontre les coups critiques !",
  "noCritOnAddEnemy": "{{moveName}} immunise l’équipe ennemie\ncontre les coups critiques !",
  "noCritOnRemove": "Les effets d’{{moveName}}\nsur {{pokemonNameWithAffix}} prennent fin !",
```

**Review by Adri1 — COMMENTED** [review]
Added French

File: src/locales/zh_CN/arena-tag.ts [hunk 4]
```diff
@@ -22,6 +22,9 @@ export const arenaTag: SimpleTranslationEntries = {
   "conditionalProtectOnAddEnemy": "{{moveName}}\n保护了敌方！",
   "conditionalProtectApply": "{{moveName}}\n保护了{{pokemonNameWithAffix}}！",
   "matBlockOnAdd": "{{pokemonNameWithAffix}}正在\n伺机使出掀榻榻米！",
+  "noCritOnAddPlayer": "The {{moveName}} shielded your\nteam from critical hits!",
+  "noCritOnAddEnemy": "The {{moveName}} shielded the opposing\nteam from critical hits!",
+  "noCritOnRemove": "{{pokemonNameWithAffix}}'s {{moveName}}\nwore off!",
```
Comment (sonnyding1) [inline src/locales/zh_CN/arena-tag.ts:27]:
```suggestion
  "noCritOnAddPlayer": "{{moveName}}保护了你的\n队伍不被击中要害！",
  "noCritOnAddEnemy": "{{moveName}}保护了对方的\n队伍不被击中要害！",
  "noCritOnRemove": "{{pokemonNameWithAffix}}的{{moveName}}\n效果消失了！",
```

File: src/locales/zh_TW/arena-tag.ts [hunk 5]
```diff
@@ -22,6 +22,9 @@ export const arenaTag: SimpleTranslationEntries = {
   "conditionalProtectOnAddEnemy": "{{moveName}} protected the\nopposing team!",
   "conditionalProtectApply": "{{moveName}} protected {{pokemonNameWithAffix}}!",
   "matBlockOnAdd": "{{pokemonNameWithAffix}} intends to flip up a mat\nand block incoming attacks!",
+  "noCritOnAddPlayer": "The {{moveName}} shielded your\nteam from critical hits!",
+  "noCritOnAddEnemy": "The {{moveName}} shielded the opposing\nteam from critical hits!",
+  "noCritOnRemove": "{{pokemonNameWithAffix}}'s {{moveName}}\nwore off!",
```
Comment (sonnyding1) [inline src/locales/zh_TW/arena-tag.ts:27]:
```suggestion
  "noCritOnAddPlayer": "{{moveName}}保護了你的\n隊伍不被擊中要害！",
  "noCritOnAddEnemy": "{{moveName}}保護了對方的\n隊伍不被擊中要害！",
  "noCritOnRemove": "{{pokemonNameWithAffix}}的{{moveName}}\n效果消失了！",
```

**Review by sonnyding1 — COMMENTED** [review]
Added zh_CN and zh_TW translations

**Comment by EnochG1** [comment]
@innerthunder 
From now Japanese is added in beta.
Could you merge latest beta and add Japanese locale files? 🙏

**Comment by innerthunder** [comment]
> @innerthunder From now Japanese is added in beta. Could you merge latest beta and add Japanese locale files? 🙏

Done.

Gonna take this out of draft since I've heard some of the translators are busy/on vacation atm, but this is still waiting on ES and IT (and now JA) translations.

**Comment by EnochG1** [comment]
IMO you don't have to get all translation. Once you add message key with english text, translators can make new PR :)

**Pull request merged**